bin/workspace_root.mli

```diff
@@ -19,8 +19,15 @@ type t =
   }
 
 val create
-  :  default_is_cwd:bool
+  :  ?from:string
+  -> default_is_cwd:bool
   -> specified_by_user:string option
-  -> (t, User_message.t) result
+  -> unit
+  -> t option
 
-val create_exn : default_is_cwd:bool -> specified_by_user:string option -> t
+val create_exn
+  :  ?from:string
```

Comment:
How about just always passing this `from` argument? I don't think there's many callers and it would help us be assured that you didn't forget to set this somewhere.